src/crypto/crypto_keygen.h

```diff
@@ -231,12 +229,14 @@ template <typename AlgorithmParams>
 struct KeyPairGenConfig final : public MemoryRetainer {
   PublicKeyEncodingConfig public_key_encoding;
   PrivateKeyEncodingConfig private_key_encoding;
-  ManagedEVPPKey key;
+  KeyObjectData key;
   AlgorithmParams params;
 
   KeyPairGenConfig() = default;
   ~KeyPairGenConfig() {
-    Mutex::ScopedLock priv_lock(*key.mutex());
+    if (key) {
```

Comment:
It seems like this needs an assertion rather than an `if` statement

Reply:
No, if the `KeyPairGenConfig` is moved, key will be empty. This shouldn't assert in that case, it just becomes a non-op.